packages/test/test-end-to-end-tests/src/test/batchBreakRegression.spec.ts

```diff
@@ -96,6 +96,7 @@ async function runAndValidateBatch(
                 documentServiceFactory: proxyDsf,
             });
         const container = await loader.resolve({ url: containerUrl });
+        await new Promise<void>((resolve) => container.once("connected", () => resolve()));
```

Comment:
Another thing to consider - This container will be in read mode. As soon as you send the first op, it will get nack'd, the container will switch to write mode and all ops that were send until then will be replayed.
It looks like this should be okay but FYI, take this into account for any validations.

Reply:
why would it matter. the test intercepts every batch sent, whether first, last, or resent. i'm worried we are just working around the error here

Reply:
It mattered before Kian's [PR that removed flush ](https://github.com/microsoft/FluidFramework/pull/11519) because there was bug where batches sent during "not connected" state did not set `batch: true` metadata on batches. Either because of the flush call and / or during replaying these ops, batches were sent just fine.

The reason I brought this up now is because even though Kian fixed that bug, in this scenario the ops go through pending state manager's replay logic. My comment was around looking at that code and making sure we are doing the right thing.

Reply:
@kian-thompson We talked about a bug where batches did not correctly end in "not connected" state. Is that fixed? Could that be causing this?

Reply:
[AB#1788](https://dev.azure.com/fluidframework/internal/_workitems/edit/1788) is not fixed yet. Not sure if the TurnBased batches getting grouped is related, but it might be? I'll try walking through with a debugger again.

Reply:
The tests use `orderSequentially` to send batches. In `TurnBased` mode, flush is not called at the end of `orderSequentially`. So, it might be worth looking at.
One consequence of this is that the ops send outside of `orderSequentially` in the same turn will be part of this batch too right? Isn't this a bug? Consider the following code in the batchBreakRegression tests. Won't the non-batch message be part of the batch in orderSequentially?
```
testObject.context.containerRuntime.orderSequentially(() => {
  for (let i = 0; i < 10; i++) {
    testObject.root.set(i.toString(), i);
  }
});
// send non-batch
testObject.root.set("foo", "bar");
```

Reply:
I validated that all ops sent before and after `orderSequentially` are also part of the same batch if they are sent in the same JS turn. It seems okay since the `orderSequentially` ops are still in the same batch but is this the intended behavior?